packages/account-wasm/src/lib.rs

```diff
@@ -183,7 +183,11 @@ impl CartridgeAccount {
     }
 
     #[wasm_bindgen(js_name = signMessage)]
-    pub fn sign_message(&self) -> Vec<JsValue> {
-        unimplemented!("Sign Message not implemented");
+    pub async fn sign_message(&self, typed_data: String) -> Result<JsValue> {
```

Comment:
should this be a string or a typed data structure?

Reply:
We could create a intermediate type TypedData for wasm but I feel like it's not necessary. Just need to go from the existing TypedData in JS to existing TypedData in rust, serde is probably the easiest